From ab0d66f33a7b3ec2ba54057d3ab3e3b5b76f4975 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Mon, 13 Nov 2017 14:30:36 -0800
Subject: [PATCH 4/6] ld/ppc/spu: Also set expld.textseg.phase to exp_seg_none

	* emultempl/ppc32elf.em (ppc_before_allocation): Also set
	expld.textseg.phase to exp_seg_none.
	* emultempl/ppc64elf.em (prelim_size_sections): Likewise.
	* emultempl/spuelf.em (spu_before_allocation): Likewise.
---
 ld/emultempl/ppc32elf.em | 1 +
 ld/emultempl/ppc64elf.em | 1 +
 ld/emultempl/spuelf.em   | 1 +
 3 files changed, 3 insertions(+)

diff --git a/ld/emultempl/ppc32elf.em b/ld/emultempl/ppc32elf.em
index f78a117222..5817fc99c0 100644
--- a/ld/emultempl/ppc32elf.em
+++ b/ld/emultempl/ppc32elf.em
@@ -149,6 +149,7 @@ ppc_before_allocation (void)
       if (expld.phase != lang_mark_phase_enum)
 	{
 	  expld.phase = lang_mark_phase_enum;
+	  expld.textseg.phase = exp_seg_none;
 	  expld.dataseg.phase = exp_seg_none;
 	  one_lang_size_sections_pass (NULL, FALSE);
 	  lang_reset_memory_regions ();
diff --git a/ld/emultempl/ppc64elf.em b/ld/emultempl/ppc64elf.em
index d6d5b0acf3..131bae7b54 100644
--- a/ld/emultempl/ppc64elf.em
+++ b/ld/emultempl/ppc64elf.em
@@ -266,6 +266,7 @@ prelim_size_sections (void)
   if (expld.phase != lang_mark_phase_enum)
     {
       expld.phase = lang_mark_phase_enum;
+      expld.textseg.phase = exp_seg_none;
       expld.dataseg.phase = exp_seg_none;
       one_lang_size_sections_pass (NULL, FALSE);
       /* We must not cache anything from the preliminary sizing.  */
diff --git a/ld/emultempl/spuelf.em b/ld/emultempl/spuelf.em
index 8d8aa25d0a..3878086c0d 100644
--- a/ld/emultempl/spuelf.em
+++ b/ld/emultempl/spuelf.em
@@ -272,6 +272,7 @@ spu_before_allocation (void)
       /* Size the sections.  This is premature, but we need to know the
 	 rough layout so that overlays can be found.  */
       expld.phase = lang_mark_phase_enum;
+      expld.textseg.phase = exp_seg_none;
       expld.dataseg.phase = exp_seg_none;
       one_lang_size_sections_pass (NULL, TRUE);
 
-- 
2.13.6

